pkg/phases/postgresoperator/test.go

```diff
@@ -143,35 +143,50 @@ func TestLogicalBackupE2E(p *platform.Platform, test *console.TestResults) {
 	test.Passf(testName, "verified test fixtures is in %s after the restore", cluster2ZalandoPsqlName)
 }
 
-func newPostgresqlDBCluster(namespace, name string) *postgresdbv1.PostgresqlDB {
-	return &postgresdbv1.PostgresqlDB{
+func newPostgresqlDBCluster(namespace, name string) *postgresdbv2.PostgresqlDB {
+	return &postgresdbv2.PostgresqlDB{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "PostgresqlDB",
-			APIVersion: "db.flanksource.com/v1",
+			APIVersion: "db.flanksource.com/v2",
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      fmt.Sprintf("e2e-test-%s-%s", utils.RandomString(6), name),
 			Namespace: namespace,
 		},
-		Spec: postgresdbv1.PostgresqlDBSpec{
-			Storage: postgresdbv1.Storage{
+		Spec: postgresdbv2.PostgresqlDBSpec{
+			Storage: postgresdbv2.Storage{
 				Size: "500Mi",
 			},
-			Backup: postgresdbv1.PostgresqlBackup{
+			Backup: postgresdbv2.PostgresqlBackup{
 				Bucket: fmt.Sprintf("logical-backup-test-%s", utils.RandomString(6)),
-				Retention: postgresdbv1.BackupRetention{
+				Retention: postgresdbv2.BackupRetention{
 					KeepLast: 5,
 				},
 			},
 			Replicas: 1,
 		},
-		Status: postgresdbv1.PostgresqlDBStatus{
+		Status: postgresdbv2.PostgresqlDBStatus{
 			Conditions: []v1.Condition{},
 		},
 	}
 }
 
-func removeLogicalBackupE2ECluster(p *platform.Platform, test *console.TestResults, db *postgresdbv1.PostgresqlDB) {
+func waitForPgClusterToReady(p *platform.Platform, namespace, postgresqlClusterName string, timeout time.Duration) error {
```

Comment:
```suggestion
// TODO: replace with platform.WaitForResource
func waitForPgClusterToReady(p *platform.Platform, namespace, postgresqlClusterName string, timeout time.Duration) error {
```

Reply:
I used `platform.WaitForResource` at first but it doesn't work as I want (to really wait until the cluster is ready to connect), and since we revert the change in kommons, I have to re-write the checking logic here to make the test work. Discussion: https://github.com/flanksource/kommons/pull/34#discussion_r610350570